File: MPP/MPP Proiect/CSharp/persistance/Interfaces/IRepository.cs
using System.Collections.Generic;
using model;

namespace persistance.Interfaces{
    public interface IRepository<ID, T> where T : Entity<ID>{
        
        void Add(T entity);

        void Remove(T entity);

        void RemoveById(int id);

        List<T> GetAll();

        T GetById(ID id);
    }
}

File: MPP/MPP Proiect/Java/Persistence/src/main/java/persistence/repository/interfaces/IRepository.java
package persistence.repository.interfaces;

import model.Entity;

import java.io.Serializable;
import java.util.ArrayList;

public interface IRepository<ID extends Serializable, T extends Entity<ID>> {

    void add(T entity);

    void remove(T entity);

    void removeById(int id);

    ArrayList<T> getAll();

    T getById(ID id);
}
